src/decisionengine/framework/taskmanager/TaskManager.py

```diff
@@ -70,13 +74,15 @@ class Worker:
     execution
     """
 
-    def __init__(self, conf_dict, base_class, channel_name):
+    def __init__(self, key, conf_dict, base_class, channel_name):
         """
         :type conf_dict: :obj:`dict`
         :arg conf_dict: configuration dictionary describing the worker
         """
         self.module_instance = _create_module_instance(conf_dict, base_class, channel_name)
         self.module = conf_dict["module"]
+        self.module_key = key
+        self.full_key = f"{channel_name}-{self.module_key}"
```

Comment:
I'd be tempted to switch this to something like `{channel_name}.{self.module_key}` so we can filter on `channel_name` down the road or add sub keys if need be.

Reply:
Done